ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/resources/KsqlResource.java

```diff
@@ -215,49 +289,19 @@ private KsqlEntity executeStatement(
                || statement instanceof CreateAsSelect
                || statement instanceof TerminateQuery
     ) {
-      if (statement instanceof AbstractStreamCreateStatement) {
-        AbstractStreamCreateStatement streamCreateStatement = (AbstractStreamCreateStatement)
-            statement;
-        Pair<AbstractStreamCreateStatement, String> avroCheckResult =
-            maybeAddFieldsFromSchemaRegistry(streamCreateStatement, streamsProperties);
-
-        if (avroCheckResult.getRight() != null) {
-          statement = avroCheckResult.getLeft();
-          statementText = avroCheckResult.getRight();
-        }
-      }
-      //Sanity check for the statement before distributing it.
-      validateStatement(statement, statementText, streamsProperties);
       return distributeStatement(statementText, statement, streamsProperties);
-    } else {
-      if (statement != null) {
-        throw new KsqlException(String.format(
-            "Cannot handle statement of type '%s'",
-            statement.getClass().getSimpleName()
-        ));
-      } else {
-        throw new KsqlException(String.format(
-            "Unable to execute statement '%s'",
-            statementText
-        ));
-      }
     }
+    // once we have distinct exception types we won't need a separate
+    // validation phase for each statement and this can go away. For now
+    // all exceptions are KsqlExceptions so we have to use thee context to
+    // decide if its an input or system error
+    throw new RuntimeException("unreachable line");
```

Comment:
Consider different message. It's not a good user experience if code gets modified and user is presented with a "unreachable line" exception ;)